python/lsst/meas/astrom/polynomialTransform.cc

```diff
@@ -0,0 +1,126 @@
+#include "lsst/meas/astrom/PolynomialTransform.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace astrom {
+
+namespace {
+
+void declarePolynomialTransform(py::module & mod) {
+    py::class_<PolynomialTransform, std::shared_ptr<PolynomialTransform>> cls(mod, "PolynomialTransform");
+
+    cls.def(py::init<ndarray::Array<double const,2,2> const &, ndarray::Array<double const,2,2> const &>(),
+            "xCoeffs"_a, "yCoeffs"_a);
+    cls.def(py::init<PolynomialTransform const &>(), "other"_a);
+
+    cls.def_static("convert",
+                   (PolynomialTransform (*)(ScaledPolynomialTransform const &)) &PolynomialTransform::convert,
+                   "other"_a);
+    cls.def_static("convert",
+                   (PolynomialTransform (*)(SipForwardTransform const &)) &PolynomialTransform::convert,
+                   "other"_a);
+    cls.def_static("convert",
+                   (PolynomialTransform (*)(SipReverseTransform const &)) &PolynomialTransform::convert,
+                   "other"_a);
+
+    cls.def("__call__", &PolynomialTransform::operator(), "in"_a, py::is_operator());
+
+    cls.def("getOrder", &PolynomialTransform::getOrder);
+    cls.def("getXCoeffs", &PolynomialTransform::getXCoeffs);
+    cls.def("getYCoeffs", &PolynomialTransform::getYCoeffs);
+    cls.def("linearize", &PolynomialTransform::linearize);
+}
+
+void declareScaledPolynomialTransform(py::module & mod) {
+    py::class_<ScaledPolynomialTransform,
+               std::shared_ptr<ScaledPolynomialTransform>> cls(mod, "ScaledPolynomialTransform");
+
+    cls.def(py::init<PolynomialTransform const &,
+                     afw::geom::AffineTransform const &,
+                     afw::geom::AffineTransform const &>(),
+            "poly"_a, "inputScaling"_a, "outputScalingInverse"_a);
+
+    cls.def(py::init<ScaledPolynomialTransform const &>(), "other"_a);
+
+    cls.def_static("convert",
+                   (ScaledPolynomialTransform (*)(PolynomialTransform const &))
+                        &ScaledPolynomialTransform::convert,
+                   "other"_a);
+    cls.def_static("convert",
+                   (ScaledPolynomialTransform (*)(SipForwardTransform const &))
+                        &ScaledPolynomialTransform::convert,
+                   "other"_a);
+    cls.def_static("convert",
+                   (ScaledPolynomialTransform (*)(SipReverseTransform const &))
+                        &ScaledPolynomialTransform::convert,
+                   "other"_a);
+
+    cls.def("__call__", &ScaledPolynomialTransform::operator(), "in"_a, py::is_operator());
+
+    cls.def("getPoly", &ScaledPolynomialTransform::getPoly, py::return_value_policy::copy);
+    cls.def("getInputScaling", &ScaledPolynomialTransform::getInputScaling, py::return_value_policy::copy);
+    cls.def("getOutputScalingInverse", &ScaledPolynomialTransform::getOutputScalingInverse,
+            py::return_value_policy::copy);
```

Comment:
In this case the returned object is intrinsically const so `reference_internal` does seem to be a better choice. I will change it.